README.md

```diff
@@ -172,245 +197,193 @@ Ipld::DagJson::serialize(
             ]),
         )
     ])
-);
+).to_json()
 ```
 
 The signature is then checked against the above fields, which passes since there's only a `role: "user"` entry. The application then uses the original JSON with the `role: "admin"` entry.
 
-# 2 Safety
+# Safety
 
-Data that has already been parsed to an in-memory IPLD representation can be canonically encoded trivially: it has already been through a [parser / validator][Parse Don't Validate].
+Data already parsed to an in-memory IPLD representation can be canonically encoded trivially: it has already been through a [parser / validator][Parse Don't Validate].
 
-Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signalled if it does not match.
+Data purporting to conform to an IPLD encoding (such as [DAG-JSON]) MUST be validated prior to signature verification. This MAY be as simple as round-trip decoding/encoding the JSON and checking that the hash matches. A validation error MUST be signaled if it does not match.
 
-> [Implementers] may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
+> Implementers may provide an opt-in for systems where round-trip determinism is a desireable [sic] feature and backward compatibility with old, non-strict data is unnecessary.
 >
-> — [DAG-JSON Spec][DAG-JSON]
+> — [DAG-JSON Spec][DAG-JSON]
 
 As it is critical for guarding against various attacks, the assumptions around canonical encoding MUST be enforced.
 
-## 2.1 Signing CIDs
+## Signing CIDs
 
-Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it is impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, disk, and networking impacts.
+Rather than validating the inline IPLD, replacing the data with a CID link to the content MAY be used instead. Note while this is very safe (as it is impractical to alter a signed hash), this approach mixes data layout with security, and may have a performance, storage, data availability, and networking consequences.
 
-### 2.1.1 Caching & Invalidation
+### Caching & Invalidation
 
 Signing CIDs has two additional caching consequences:
 
 1. Signing CIDs enables a simple strategy for caching validation by CID.
 2. Such a strategy also MAY require accounting for revocation of the signing keys themselves. In this case, the cache would need to include additional information about the signing key.
 
-## 2.2 Raw (Noncanonicalized) Data
+# Format
```

Comment:
everything above this section IMO should be removed! IPLD, CID, etc, etc 
this spec SHOULD not care about any of it

Reply:
This spec is explicitly about how to sign over IPLD data. It would be called "multisig" except that's the name of an existing concept (namespace conflict). Part of the motivation is that we can't sign over IPLD data without knowing the serialisation and whatnot. If someone wants to sign non-IPLD data, they can use the raw encoding prefix.

Reply:
it has that benefit of helping ipld data signatures but not restrict to it.
we talk about EIP191 and JWT in which the payload is not ipld data (or may not be).

>If someone wants to sign non-IPLD data, they can use the raw encoding prefix.

what does this mean? itsnt varsig just a bunch of prefixes?

On the other hand, the non-IPLD data schemes we want to support are strange to combine with varsig,

- JWT: i dont even know how to do JWT with varsig cause JWT is kinda similar to varsig just a json header 
- EIP191: you are supposed to sign UTF8 string using personal_sign but you can encode varsig+payload and it works
- webauthn: still need to recheck this but im guessing we will put varsig+payload in the challenge as a string or a CID